Error: Could not resolve dependencies for latest snapshot

I'm trying to use the latest App Insights Spring Boot Starter in my project, but Maven does not seem to be able to fetch it from the repository:

I would like to use: 
[](https://mvnrepository.com/artifact/com.microsoft.azure/applicationinsights-spring-boot-starter/2.6.0-BETA.2)

So my POM looks like:

```
<!-- https://mvnrepository.com/artifact/com.microsoft.azure/applicationinsights-spring-boot-starter -->
<dependency>
    <groupId>com.microsoft.azure</groupId>
    <artifactId>applicationinsights-spring-boot-starter</artifactId>
    <version>2.6.0-BETA.2</version>
</dependency>
```

While running `mvn clean install -U` - I get:

`[ERROR] Failed to execute goal on project <whatever>: Could not resolve dependencies for project com.example:whatever:jar:0.0.1-SNAPSHOT: Could not find artifact com.microsoft.azure:applicationinsights-web:jar:all:2.6.0-BETA.2 in central (https://repo.maven.apache.org/maven2) ->`

So I tried adding a custom repository for : `https://repo1.maven.org/maven2/` - something like:

```
 <repositories>
    <repository>
      <id>whatever</id>
      <url>https://repo1.maven.org/maven2</url>
    </repository>
  </repositories>
```
That also doesn't work:

`[ERROR] Failed to execute goal on project jvwapp: Could not resolve dependencies for project com.example:jvwapp:jar:0.0.1-SNAPSHOT: Could not find artifact com.microsoft.azure:applicationinsights-web:jar:all:2.6.0-BETA.2 in whatever (https://repo1.maven.org/maven2) -> [`

Stuff like this does not seem to work either:

```
  <repositories>
    <repository>
      <id>whatever</id>
      <url>https://repo1.maven.org/maven2</url>
      <releases>
        <enabled>false</enabled>
      </releases>
      <snapshots>
        <enabled>true</enabled>
      </snapshots>
    </repository>
  </repositories>
```
Any suggestions?

--- a/azure-application-insights-spring-boot-starter/build.gradle
+++ b/azure-application-insights-spring-boot-starter/build.gradle
@@ -95,7 +95,14 @@ whenPomConfigured = { p ->
             dep.artifactId != agentArtifactId && dep.artifactId != loggerArtifactId &&
             !(dep.groupId in ['org.apache.http', 'eu.infomas', 'org.apache.commons', 'commons-io',
                               'com.google.guava', 'com.google.code.gson', 'org.apache.httpcomponents',
-                              'io.grpc', 'com.google.protobuf'])}
+                              'io.grpc', 'com.google.protobuf'])
+    }
+    .collect { dep ->
+        if (dep.artifactId == 'applicationinsights-web') {
+            dep.classifier = null
+        }
+        dep
+    }
     p.dependencies += project.configurations.provided.allDependencies
             .findAll { it.group != 'com.microsoft.azure' }
             .collect {

